src/sqlfluff/rules/structure/ST10.py

```diff
@@ -0,0 +1,71 @@
+"""Implementation of Rule ST10."""
+
+from typing import Iterator, Tuple
+
+from sqlfluff.core.rules import BaseRule, EvalResultType, LintResult, RuleContext
+from sqlfluff.core.rules.crawlers import SegmentSeekerCrawler
+
+
+class Rule_ST10(BaseRule):
+    """Const expression rule.
+
+    **Anti-pattern**
+
+    .. code-block:: sql
+
+         -- TODO
+        SELECT 1;
+
+    **Best practice**
+
+    .. code-block:: sql
+
+         -- TODO
+        SELECT 1;
+
+    """
+
+    name = "structure.const_expression"
```

Comment:
I think the name here isn't obvious. Maybe something like `constant_condition` might be clearer?

```suggestion
    name = "structure.constant_condition"
```

Reply:
It's any expression, not necesarily a condition.

```sql
SELECT col1 = col1 AS are_equal
FROM tbl
```

This will (and should) trigger warning.

I'll add that to test cases.